internal/provider/common_helpers.go

```diff
@@ -46,3 +46,13 @@ func skipNotCI(t *testing.T) {
 		t.Skip("Skipping testing in CI environment")
 	}
 }
+
+func getTenantId() string {
```

Comment:
There should be a way to access the Provider instance and get the tenant from our own provider instance struct - can you try that path? If you have any difficulty, let's look into it together.